component/core/src/main/java/org/exoplatform/social/core/metadata/attachment/AttachmentServiceImpl.java

```diff
@@ -128,23 +128,72 @@ public ObjectAttachmentOperationReport createAttachments(ObjectUploadResourceLis
+  }
+
+  @Override
+  public ObjectAttachmentOperationReport updateAttachments(FileAttachmentResourceList attachmentList,
+                                                           Identity userAclIdentity) throws ObjectNotFoundException,
+                                                                                     IllegalArgumentException,
+                                                                                     IllegalAccessException {
+
+    if (attachmentList == null) {
+      throw new IllegalArgumentException("Attachment is mandatory");
+    }
+
+    long userIdentityId = attachmentList.getUserIdentityId();
+    if (userIdentityId <= 0) {
+      throw new IllegalArgumentException("User identity id is mandatory");
+    }
+
+    org.exoplatform.social.core.identity.model.Identity userIdentity =
+                                                                     identityManager.getOrCreateUserIdentity(userAclIdentity.getUserId());
+
+    if (userIdentity == null || userIdentity.isDeleted() || !userIdentity.isEnable()) {
+      throw new IllegalStateException("User with id " + userAclIdentity.getUserId() + " isn't valid");
+    }
+    String objectType = attachmentList.getObjectType();
+    if (StringUtils.isBlank(objectType)) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    String objectId = attachmentList.getObjectId();
+    if (StringUtils.isBlank(objectId)) {
+      throw new IllegalArgumentException("Object identifier is mandatory");
+    }
+
+    MetadataObject metadataObject = new MetadataObject(objectType, objectId);
+
+    List<MetadataItem> existingAttachments =
+                                           metadataService.getMetadataItemsByMetadataTypeAndObject(AttachmentService.METADATA_TYPE.getName(),
+                                                                                                   metadataObject);
+    if (existingAttachments.isEmpty()) {
+      List<String> existingFileIds = existingAttachments.stream()
+                                                        .map(existingAttachment -> existingAttachment.getMetadata().getName())
+                                                        .toList();
+      List<String> remainingFileIds = attachmentList.getFileIds();
+      if (remainingFileIds.isEmpty()) {
+        for (String fileId : existingFileIds) {
+          deleteAttachment(objectType, objectId, fileId, userAclIdentity);
+        }
+      } else {
+        List<String> fileToDelete = remainingFileIds.stream().filter(fileId -> !existingFileIds.contains(fileId)).toList();
+        for (String fileId : fileToDelete) {
+          deleteAttachment(objectType, objectId, fileId, userAclIdentity);
+        }
+      }
+    }
+
+    String parentObjectId = attachmentList.getParentObjectId();
+
+    List<String> uploadIds = attachmentList.getUploadIds();
     ObjectAttachmentOperationReport report = new ObjectAttachmentOperationReport();
```

Comment:
You may not need to instantiate this here, since you will replace it

```suggestion
    ObjectAttachmentOperationReport report = null;
```